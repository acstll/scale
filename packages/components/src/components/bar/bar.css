.bar {
  box-sizing: border-box;
  padding: 0 1rem;
  overflow: hidden;
  display: flex;
  position: sticky;
  top: 0;
  left: 0;
  right: 0;
  background: #e20074;
  color: white;
  z-index: 999;
  margin: 0;
  width: 100vh;
}

.bar__body {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.5rem 0;
  width: 100vh;
  margin: 0;
}

.bar__scrolled {
  position: fixed;
  padding: 0;
  margin: 0;
  animation: collapse 3s forwards ease-in;
}
@keyframes collapse {
  from {
    height: 50px;
  }
  to {
    height: 5px;
  }
}

.bar__body .shrink ::slotted(*) {
  transition: all 1s ease;
  height: 5px;
}

.shrink {
  height: 5px;
  transition: all 1s ease;
}
