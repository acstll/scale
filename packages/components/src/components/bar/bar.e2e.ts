import { newE2EPage } from '@stencil/core/testing';

describe('t-bar', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-bar />');
    const element = await page.find('t-bar');
    expect(element).toHaveClass('hydrated');
  });
});
