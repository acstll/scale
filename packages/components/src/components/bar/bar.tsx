import { Component, Prop, h, Listen, Element, Method } from '@stencil/core';

import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { HTMLStencilElement } from '@stencil/core/internal';

@Component({
  tag: 't-bar',
  styleUrl: 'bar.css',
  shadow: true,
})
export class Bar {
  @Element() public el: HTMLStencilElement;
  /** (optional) Bar size */
  @Prop() public size?: string = '';
  /** (optional) Bar theme */
  @Prop() public theme?: string = '';
  /** (optional) Bar variant */
  @Prop() public variant?: string = '';
  @Prop({ reflectToAttr: true }) public open?: boolean = false;

  // private showSlots: boolean;

  // public componentDidLoad() {
  //   this.showSlots = !!this.el.querySelector('[slot="logo"]');
  // }

  @Listen('scroll', { target: 'window' })
  public handleScroll() {
    // console.log('scrolled', e);
  }

  @Method()
  public async openBar() {
    this.open = true;
  }

  public render() {
    if (!this.open) {
      return null;
    }
    return (
      <bar class={this.getCssClassMap()}>
        <div class="bar__body">
          <slot name="logo" />
          <slot name="claim" />
        </div>
      </bar>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'bar',
      this.size && `bar--size-${this.size}`,
      this.theme && `bar--theme-${this.theme}`,
      this.variant && `bar--variant-${this.variant}`
    );
  }
}
