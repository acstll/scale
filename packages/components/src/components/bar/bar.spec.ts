import { newSpecPage } from '@stencil/core/testing';
import { Bar } from './bar';

describe('Bar', () => {
  let element;
  beforeEach(async () => {
    element = new Bar();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Bar],
      html: `<t-bar/>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle size css class', () => {
    element.size = 'small';
    expect(element.getCssClassMap()).toContain('bar--size-small');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('bar--theme-default');
  });

  it('should handle variant css class', () => {
    element.variant = 'primary';
    expect(element.getCssClassMap()).toContain('bar--variant-primary');
  });
});
