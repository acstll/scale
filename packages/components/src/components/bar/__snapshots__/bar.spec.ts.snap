// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar should match snapshot 1`] = `
<t-bar>
  <mock:shadow-root>
    <bar class="bar">
      <div class="bar__body">
        <slot name="logo"></slot>
        <slot name="claim"></slot>
      </div>
    </bar>
  </mock:shadow-root>
</t-bar>
`;
